includes/Commerce/Orders.php

```diff
@@ -298,12 +299,82 @@ public function fulfill_order( \WC_Order $order, $tracking_number, $carrier ) {
 	 * @since 2.1.0-dev.1
 	 *
 	 * @param \WC_Order_Refund $refund order refund object
-	 * @param mixed $args
+	 * @param string $reason_code refund reason code
 	 * @throws SV_WC_Plugin_Exception
 	 */
-	public function add_order_refund( \WC_Order_Refund $refund, $args ) {
+	public function add_order_refund( \WC_Order_Refund $refund, $reason_code ) {
 
-		// TODO: implement
+		$plugin = facebook_for_woocommerce();
+
+		$api = $plugin->get_api( $plugin->get_connection_handler()->get_page_access_token() );
+
+		$valid_reason_codes = [
+			Refund_Request::REASON_BUYERS_REMORSE,
+			Refund_Request::REASON_DAMAGED_GOODS,
+			Refund_Request::REASON_NOT_AS_DESCRIBED,
+			Refund_Request::REASON_QUALITY_ISSUE,
+			Refund_Request::REASON_OTHER,
+			Refund_Request::REASON_WRONG_ITEM,
+		];
+
+		if ( ! in_array( $reason_code, $valid_reason_codes, true ) ) {
+			$reason_code = Refund_Request::REASON_OTHER;
+		}
+
+		try {
+
+			$parent_order = wc_get_order( $refund->get_parent_id() );
+
+			if ( ! $parent_order instanceof \WC_Order ) {
+				throw new SV_WC_Plugin_Exception( __( 'Parent order not found.', 'facebook-for-woocommerce' ) );
+			}
+
+			$remote_id = $parent_order->get_meta( self::REMOTE_ID_META_KEY );
+
+			if ( ! $remote_id ) {
+				throw new SV_WC_Plugin_Exception( __( 'Remote ID for parent order not found.', 'facebook-for-woocommerce' ) );
+			}
+
+			$items = [];
+
+			/** @var \WC_Order_Item_Product $item */
+			foreach ( $refund->get_items() as $item ) {
+
+				if ( $product = $item->get_product() ) {
+
+					$items[] = [
+						'retailer_id'          => \WC_Facebookcommerce_Utils::get_fb_retailer_id( $product ),
+						'item_refund_quantity' => $item->get_quantity(),
+					];
+				}
+			}
+
+			if ( empty( $items ) ) {
+				throw new SV_WC_Plugin_Exception( __( 'No valid Facebook products were found.', 'facebook-for-woocommerce' ) );
+			}
+
+			$refund_data = [
+				'items'       => $items,
+				'reason_code' => $reason_code,
+				'reason_text' => $refund->get_reason(),
+				'shipping'    => [
+					'amount'   => $refund->get_shipping_total(),
+					'currency' => $refund->get_currency(),
```

Comment:
According to the doc examples, this needs to be wrapped in a `shipping_refund` key.

Reply:
Fixed in https://github.com/facebookincubator/facebook-for-woocommerce/pull/1525/commits/a5d4690d9d39acc799125b3a46dd132914158219.